app/repositories/index.js

```diff
@@ -25,10 +26,12 @@ async function getById(mapDataset) {
 
 async function fetchById(mapDataset) {
   try {
-    const response = await fetch(external[mapDataset]);
+    const response = await fetch(`${external[mapDataset].url}?${external[mapDataset].query}`);
     const dataset = await response.json();
 
-    return dataset;
+    const modifiedDataset = addAuthor(dataset, external[mapDataset].author);
```

Comment:
If each dataset needs to be modified differently this might be a better choice.
```javascript
return external[mapDataset].modifyDataset
      ? external[mapDataset].modifyDataset(dataset)
      : dataset;
```